core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -434,6 +437,44 @@ export class NavigationComponent {
   onClickGetAllVersions() {
     this.workflowVersionService.clickDisplayWorkflowVersions();
   }
+
+  private handleVersionDisplay(): void {
+    this.workflowVersionService
+      .getDisplayParticularVersionStream()
+      .pipe(untilDestroyed(this))
+      .subscribe(displayVersionFlag => {
+        this.particularVersionDate =
+          this.workflowActionService.getWorkflowMetadata().creationTime === undefined
+            ? ""
+            : "" +
+              this.datePipe.transform(
+                this.workflowActionService.getWorkflowMetadata().creationTime,
+                "MM/dd/yyyy HH:mm:ss zzz",
+                Intl.DateTimeFormat().resolvedOptions().timeZone,
+                "en"
+              );
+        this.displayParticularWorkflowVersion = displayVersionFlag;
+      });
+  }
+
+  closeParticularVersionDisplay() {
+    this.displayParticularWorkflowVersion = false;
+    this.workflowActionService.enableWorkflowModification();
+    // reload the old workflow don't persist anything
+    this.workflowActionService.reloadWorkflow(this.workflowActionService.getTempWorkflow());
+    this.workflowActionService.resetTempWorkflow();
```

Comment:
you might want to be careful with `redo` `undo` upon reloading a workflow. Also are those two methods out of order? I assume we should reset temp workflow then reload the original workflow back. please correct me if I am wrong.

Reply:
the order is correct. I need to get it first before resetting it.
and nice catch regarding `redo` `undo` stack. I will investigate it